utils/apply-tag.ts

```diff
@@ -0,0 +1,32 @@
+import { IAspect, ITaggable, TagManager } from 'aws-cdk-lib';
+import { IConstruct } from 'constructs';
+
+export type Tags = { [key: string]: string } & {
+    environment: 'development' | 'staging' | 'production' | 'demonstration';
+    project: string;
+    owner: string;
+};
+
+export class ApplyTags implements IAspect {
+  #tags: Tags;
+
+  constructor(tags: Tags) {
+    this.#tags = tags;
+  }
+
+  visit(node: IConstruct) {
+    if (TagManager.isTaggable(node)) {
+      Object.entries(this.#tags).forEach(([key, value]) => {
+        console.log(`Applying tag ${key}=${value} to ${node.node.path}`);
```

Comment:
**Suggestion:** Remove the console log statement inside the `visit` method to avoid unnecessary logging in production.
> 👑  **Medium Priority** 
 > [category: maintainability, importance: 6]
> **Reason:** Removing console logs from production code improves cleanliness and performance, although it's a relatively minor enhancement.
 

```suggestion
        // console.log(`Applying tag ${key}=${value} to ${node.node.path}`);
```